### ISSUE TITLE
Form validation, consider zero (0) as a value for required fields

### ISSUE THREAD

**Issue by tripflex** [Automattic/WP-Job-Manager#1480]
Consider the example, you create a field `job_salary` and configure it in PHP as a **required** field, to require user to enter a value.

If you wanted the user to be able to enter a value of 0 (zero), the form validation returns an error, because PHP's `empty` considers 0 (zero) and empty value.

http://php.net/manual/en/function.empty.php

https://github.com/Automattic/WP-Job-Manager/blob/2a7e33075ce935d6ff7719df3437a2c28f0c0c7e/includes/forms/class-wp-job-manager-form-submit-job.php#L305

My thought was to maybe create a helper function/method to replace using `empty` and maybe pass through a filter of "false" values to check against?

```php
protected function is_empty( $value ){
    //  @see http://php.net/manual/en/function.empty.php -- standard default empty values
    $false_vals = apply_filters( 'submit_job_form_validate_fields_empty_values', array( '', 0, 0.0, '0', null, false, array() );

    // strict true for type checking also
    if( in_array( $value, $false_vals, true ) ){
        return true;
    }

    return false;
}
```

Then if end user really wants to require a value but still consider 0 (zero) a true value, it's as easy as just using that filter to remove it from the array ... thoughts?

**Comment by tripflex** [comment]
@jom thoughts on this?  I can submit a PR for this, if anything just to allow a filter to not rely specifically on php's `empty` which considers a lot of different false values

**Comment by jom** [comment]
Sounds like a sensible solution!

**Comment by tripflex** [comment]
Just to recap our conversation @jom with the new refactoring:
https://github.com/Automattic/wp-job-manager/blob/9cdc98d673c830512f7f435a77a135b71e78e69a/includes/abstracts/abstract-wp-job-manager-form.php#L437

Can just come up with a snippet to add a `before_sanitize` callback to set the `empty` parameter on the field, allowing an empty value to not return generic "x is a required field"

Could then easily also add a filter on `submit_job_form_validate_fields` to validate the field even further, to check if non-numeric value is passed, and then provide a valid user helpful error (instead of generic one)

**Comment by sanaazmeh** [comment]
I think it should be  as following:
` if ( $field['required'] &&  ($values[ $group_key ][ $key ] =='' ||  $values[ $group_key ][ $key ] == null )) { `

**Comment by aaronfc** [comment]
Hi @tripflex, this was finally merged in `1.35.0`, right?
Is there anything pending from your side or we can close this issue?

**Comment by wgicio** [comment]
would be great to get this fixed up
